include/swift/AST/DiagnosticsSema.def

```diff
@@ -2446,6 +2446,12 @@ WARNING(differentiable_implicit_noderivative_fixit,none,
       "stored property %0 has no derivative because it does not conform to "
       "'Differentiable'; add '@noDerivative' to make it explicit",
       (Identifier))
+WARNING(differentiable_constant_property_implicit_noderivative_fixit,none,
+      "'let' properties with an initial value do not have a derivative; add "
+      "'@noDerivative' to make it explicit",
```

Comment:
"initial value" -> "default value"

From https://docs.swift.org/swift-book/LanguageGuide/Properties.html:
> You can provide a default value for a stored property as part of its definition, as described in Default Property Values. You can also set and modify the initial value for a stored property during initialization. This is true even for constant stored properties, as described in Assigning Constant Properties During Initialization.

Reply:
Nice! Done in https://github.com/apple/swift/pull/22905/commits/eb580db59d15b7dbe98178f350d680a59846a95b.

Reply:
Note in this paragraph the last sentence is slightly confusing -- it seems to suggest overriding a default `let` with an initializer is possible.

Reply:
Yes, that is confusing.
I wonder if the semantics are changed by [SE-0242](https://forums.swift.org/t/se-0242-synthesize-default-values-for-the-memberwise-initializer/20618?u=dan-zheng).